### PR TITLE
add settable method to rcu API

### PR DESCRIPTION
**Pull request by sheharyaar**

### CODE REVIEW

File: lib/luarcu.c [hunk 1]
```diff
@@ -177,6 +177,19 @@ lunatik_object_t *luarcu_gettable(lunatik_object_t *table, const char *key, size
 }
 EXPORT_SYMBOL(luarcu_gettable);
 
+int luarcu_settable(lunatik_object_t *table, const char *key, size_t keylen, lunatik_object_t *object)
+{
+	int ret;
+
+	lunatik_lock(table);
+	ret = luarcu_insert(table->private, key, keylen, object);
```
Comment (lneto) [inline lib/luarcu.c:185]:
```suggestion
	ret = luarcu_insert((luarcu_table_t *)table->private, key, keylen, object);
```

File: lib/luarcu.c [hunk 2]
```diff
@@ -203,14 +216,8 @@ static int luarcu_newindex(lua_State *L)
...
-	int ret;
-
-	lunatik_lock(table);
-	ret = luarcu_insert(table->private, key, keylen, object);
-	lunatik_unlock(table);
-	if (!rcu_read_lock_held())
-		synchronize_rcu();
 
+	int ret = luarcu_settable(table, key, keylen, object);
 	if (ret != 0)
```
Comment (lneto) [inline lib/luarcu.c:221]:
```suggestion
	if (luarcu_settable(table, key, keylen, object) < 0)
```

File: lib/luarcu.c [hunk 3]
```diff
@@ -177,6 +177,19 @@ lunatik_object_t *luarcu_gettable(lunatik_object_t *table, const char *key, size
...
+int luarcu_settable(lunatik_object_t *table, const char *key, size_t keylen, lunatik_object_t *object)
+{
+	int ret;
+
+	lunatik_lock(table);
+	ret = luarcu_insert(table->private, key, keylen, object);
+	lunatik_unlock(table);
+	if (!rcu_read_lock_held())
+		synchronize_rcu();
+	return ret;
```
Comment (lneto) [inline lib/luarcu.c:189]: I think returning `-ENOMEM` would be more meaningful for the API client.. perhaps the best place to add it is here:

https://github.com/luainkernel/lunatik/pull/118/files#diff-87fd494e0e522758b53f3d5ec4a3c8e9886f1631f5ed1efb61a56755a22ba37dR115

and here:

https://github.com/luainkernel/lunatik/pull/118/files#diff-87fd494e0e522758b53f3d5ec4a3c8e9886f1631f5ed1efb61a56755a22ba37dR120

**Review by lneto — CHANGES_REQUESTED** [review]
quite minor, I would rename your commit based on this commit message.. this is not a method but a C API..

https://github.com/luainkernel/lunatik/commit/c586e5ddf7961e2230ca0d5830eedcc859932a0e

e.g., "add luarcu_settable() C API"

**Comment by sheharyaar** [comment]
Done

**Review by lneto — APPROVED** [review]

**Pull request merged**